### PR TITLE
MYST-224 Refactor `identityHandler`, add tests

### PR DESCRIPTION
**Pull request by donce**
I'm going to work around `identityHandler` for `MYST-224`, so it should have at least basic tests.

### CODE REVIEW

File: cmd/mysterium_server/command_run/factory.go [hunk 1]
```diff
@@ -73,3 +75,15 @@ func NewCommandWith(
 		},
 	}
 }
+
+func SelectIdentity(identityHandler *identityHandler, keyOption string) (id identity.Identity, err error) {
```
Comment (Waldz) [inline cmd/mysterium_server/command_run/factory.go]: Idea of handler was, to have thiner code in command.
This why, I moved it to `identity_handler.go` next to `command.go`.

File: cmd/mysterium_server/command_run/identity_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import (
+	"testing"
+
+	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/server"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_identityHandler_UseExisting(t *testing.T) {
```
Comment (Waldz) [inline cmd/mysterium_server/command_run/identity_handler_test.go]: +100

File: cmd/mysterium_server/command_run/identity_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	identityManager := identity.NewIdentityManagerFake()
+	client := server.NewClientFake()
+	cache := identity.NewIdentityCacheFake()
+
+	handler := NewNodeIdentityHandler(identityManager, client, cache)
+
+	id, err := handler.UseExisting("address")
```
Comment (Waldz) [inline cmd/mysterium_server/command_run/identity_handler_test.go]: But such value "address" is not mocked in `identityManager`. is not it?

File: cmd/mysterium_server/command_run/identity_handler_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+func Test_identityHandler_UseNew(t *testing.T) {
+	identityManager := identity.NewIdentityManagerFake()
+	client := server.NewClientFake()
+	cache := identity.NewIdentityCacheFake()
+
+	handler := NewNodeIdentityHandler(identityManager, client, cache)
+
+	id, err := handler.UseNew()
+	assert.Equal(t, identityManager.FakeIdentity2, id)
```
Comment (Waldz) [inline cmd/mysterium_server/command_run/identity_handler_test.go]: You should assert if it was registered via `client`

File: identity/cache.go [hunk 5]
```diff
@@ -15,23 +16,24 @@ type IdentityCache struct {
...
 }
 
 func (ic *IdentityCache) GetIdentity() (identity Identity, err error) {
-	if ic.cacheExists() {
-		cache, err := ic.readCache()
-		if err != nil {
-			return identity, err
-		}
+	if !ic.cacheExists() {
+		err = errors.New("cache file does not exist")
```
Comment (Waldz) [inline identity/cache.go]: It's not an error if file is missing. It's same as "identity non found in cache"

File: cmd/mysterium_server/command_run/factory.go (hunk 1 above)
Comment (donce) [reply cmd/mysterium_server/command_run/factory.go]: well, `SelectIdentity` is independent function - maybe it's worth to move it to new file to make less code in this file?

File: cmd/mysterium_server/command_run/identity_handler_test.go (hunk 3 above)
Comment (donce) [reply cmd/mysterium_server/command_run/identity_handler_test.go]: This address is ignored in fake identity manager:
```go
func (fakeIdm *idmFake) GetIdentity(_ string) (Identity, error) {
	return fakeIdm.FakeIdentity1, nil
}
```

File: identity/cache.go (hunk 5 above)
Comment (donce) [reply identity/cache.go]: 1. This method is named `GetIdentity` and it seems a common sense to get an identity or an error.

2. Looking in `identity_handler.go`:
```go
	identity, err = ih.cache.GetIdentity()
	if err != nil || !ih.manager.HasIdentity(identity.Address) {
		return identity, errors.New("identity not found in cache")
	}
```
If cache file does not exist, this results in "identity not found in cache" error either way - with or without this change.
What this change is about is not only making method more predictable, but also avoiding calling
`ih.manager.HasIdentity("")` for empty identity, which is returned when cache does not exist. If we know that cache file does not exist, we should not check for an empty identity in identity manager.

File: cmd/mysterium_server/command_run/identity_handler_test.go (hunk 4 above)
Comment (donce) [reply cmd/mysterium_server/command_run/identity_handler_test.go]: Done.

File: cmd/mysterium_server/command_run/identity_handler_test.go (hunk 3 above)
Comment (Waldz) [reply cmd/mysterium_server/command_run/identity_handler_test.go]: But how can we use *existing* "address" if it dont exist actually. Could could double check, plz

File: cmd/mysterium_server/command_run/factory.go (hunk 1 above)
Comment (Waldz) [reply cmd/mysterium_server/command_run/factory.go]: Clear, decoupled from from handler struct.
This separate function could still be in `node_handler.go`

File: cmd/mysterium_server/command_run/identity_handler_test.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	client := server.NewClientFake()
+	cache := identity.NewIdentityCacheFake()
+
+	fakeIdentity := identity.FromAddress("abc")
+	cache.StoreIdentity(fakeIdentity)
+
+	handler := NewNodeIdentityHandler(identityManager, client, cache)
+
+	id, err := handler.UseLast()
+	assert.Equal(t, fakeIdentity, id)
```
Comment (Waldz) [inline cmd/mysterium_server/command_run/identity_handler_test.go]: Could your assert that no identity was registered?

File: cmd/mysterium_server/command_run/identity_handler_test.go [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+func Test_identityHandler_UseExisting(t *testing.T) {
+	identityManager := identity.NewIdentityManagerFake()
+	client := server.NewClientFake()
+	cache := identity.NewIdentityCacheFake()
+
+	handler := NewNodeIdentityHandler(identityManager, client, cache)
+
+	id, err := handler.UseExisting("address")
+	assert.Equal(t, identityManager.FakeIdentity1, id)
```
Comment (Waldz) [inline cmd/mysterium_server/command_run/identity_handler_test.go]: Could your assert that no identity was registered?

File: cmd/mysterium_server/command_run/identity_handler_test.go (hunk 3 above)
Comment (donce) [reply cmd/mysterium_server/command_run/identity_handler_test.go]: Double checked - everything seems ok, not sure what's confusing you.

File: cmd/mysterium_server/command_run/factory.go (hunk 1 above)
Comment (donce) [reply cmd/mysterium_server/command_run/factory.go]: Moved it back to `node_handler` as a function.

File: cmd/mysterium_server/command_run/identity_handler_test.go (hunk 6 above)
Comment (donce) [reply cmd/mysterium_server/command_run/identity_handler_test.go]: Done.

File: cmd/mysterium_server/command_run/identity_handler_test.go (hunk 7 above)
Comment (donce) [reply cmd/mysterium_server/command_run/identity_handler_test.go]: Done.

**Comment by donce** [comment]
All comments done.

**Pull request merged**